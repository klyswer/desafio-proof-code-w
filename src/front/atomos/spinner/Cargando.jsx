import React from 'react';
import { ReactComponent as Icon } from './icon.svg';
import './cargando.css';

// ************************************* COMPONENTE (Cargando.jsx) LEER: DocumentaciÃ³n *****************
// ----------------------------------------------------------------------------------
// ---> (PROPS)
// ---> mensaje: este props recibe un "string" para indicar al usuario lo que sucede. tipo-->(STRING)
// ---> svg: este props recibe un svg, para renderizar en el componente.
// ---> girando= este props controla la clase que mueve circularmente la imagen svg. tipo-->(BOOL)
// ---> css= este props recibe una clase CSS3 en string. tipo-->(STRING)
// *********************************************************************************************************

const Cargando = ({mensaje='', svg='', girando=false, css}) => {
  return (
    <div className="row mt-4">
      <div className={`container cargando_css ${!!css && css}`}>
        {svg && <img className={girando && 'anim_girar'} src={svg===''?svg:Icon} alt="icon" /> }
        <div className="text_component_carga">{mensaje}</div>
      </div>
    </div>
  );
}

export default Cargando;
