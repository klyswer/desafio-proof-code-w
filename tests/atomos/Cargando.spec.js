import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import Cargando from '../../src/front/atomos/spinner/Cargando';

describe('<Cargando />', () => {

  const mensaje='Por favor, espere.';
  const svg=true; // Valor por defecto renderiza un spinner sencillo del componente.
  const girando=true;
  const css="clase_prueba";

  const wrapper = shallow( <Cargando mensaje={mensaje} svg={svg} girando={girando} css={css} />);

  test('Primer SnapShot: <Cargando />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Caso prueba 1: validación de "mensaje" en componente.', () => {
    const mensajeEsperado = wrapper.find('.text_component_carga').text();
    expect(mensajeEsperado).toBe(mensaje);
  });


  test('Caso prueba 2: validación de "girando" en componente agrega la clase "anim_girar".', () => {
    const mensajeEsperado = wrapper.find('img').prop('className'); 
    expect(mensajeEsperado.includes('anim_girar')).toBeTruthy();
  });

  test('Caso prueba 3: validación de "css" en componente agrega la clase "anim_girar".', () => {
    const cssEsperado = wrapper.find('.container').prop('className'); 
    expect(cssEsperado.includes(css)).toBeTruthy();
  });
});