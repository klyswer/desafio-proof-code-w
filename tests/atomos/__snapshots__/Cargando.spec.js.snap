// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cargando /> Primer SnapShot: <Cargando /> 1`] = `
<div
  className="row mt-4"
>
  <div
    className="container cargando_css clase_prueba"
  >
    <img
      alt="icon"
      className="anim_girar"
      src={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
    />
    <div
      className="text_component_carga"
    >
      Por favor, espere.
    </div>
  </div>
</div>
`;
